Anlab.Mvc/Services/DataSigningService.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using System.Threading.Tasks;
+using AnlabMvc.Models.Configuration;
+using Microsoft.Extensions.Options;
+using Serilog;
+
+namespace AnlabMvc.Services
+{
+    public interface IDataSigningService
+    {
+        string Sign(IDictionary<string, string> paramsArray);
+        bool Check(IDictionary<string, string> paramsArray, string signature);
+    }
+
+    public class DataSigningService : IDataSigningService
+    {
+        private readonly CyberSourceSettings _cyberSourceSettings;
+
+        public DataSigningService(IOptions<CyberSourceSettings> cyberSourceSettings)
+        {
+            _cyberSourceSettings = cyberSourceSettings.Value;
+        }
+
+        public string Sign(IDictionary<string, string> paramsArray)
+        {
+            return Sign(BuildDataToSign(paramsArray), _cyberSourceSettings.SecretKey);
+        }
+
+        public bool Check(IDictionary<string, string> paramsArray, string signature)
+        {
+            try
+            {
+                return signature == Sign(BuildDataToSign(paramsArray), _cyberSourceSettings.SecretKey);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, ex.Message);
+                return false;
+            }
+        }
+
+        private static string Sign(string data, string secretKey)

```

Comment:
@jSylvestre maybe give this a slightly different name, SignData() or something, because otherwise I think it's confusing that it has the same name as a public interface method on the same class